deepmd/pt/utils/dataset.py

```diff
@@ -575,44 +24,22 @@ def __init__(
         - type_map: Atom types.
         """
         self._type_map = type_map
-        if not isinstance(rcut, list):
-            if isinstance(sel, int):
-                sel = [sel]
-            sec = torch.cumsum(torch.tensor(sel), dim=0)
-        else:
-            sec = []
-            for sel_item in sel:
-                if isinstance(sel_item, int):
-                    sel_item = [sel_item]
-                sec.append(torch.cumsum(torch.tensor(sel_item), dim=0))
-        self._data_system = DeepmdDataSystem(
-            system,
-            rcut,
-            sec,
-            type_map=self._type_map,
-            type_split=type_split,
-            noise_settings=noise_settings,
-            shuffle=shuffle,
+        self._data_system = DeepmdData(
+            sys_path=system, shuffle_test=shuffle, type_map=self._type_map
         )
+        self._data_system.add("energy", 1, atomic=False, must=False, high_prec=True)
+        self._data_system.add("force", 3, atomic=True, must=False, high_prec=False)
+        self._data_system.add("virial", 9, atomic=False, must=False, high_prec=False)
         self.mixed_type = self._data_system.mixed_type
         self._ntypes = self._data_system.get_ntypes()
-        self._natoms = self._data_system._natoms
+        self._natoms = self._data_system.natoms
```

Comment:
The public method is preferred
```suggestion
        self._natoms = self._data_system.get_natoms()
```